### PR TITLE
[treescript] Add android_l10n_import and android_l10n_sync actions

### PR DESCRIPTION
**Pull request by gbrownmozilla**
This is a treescript-based replacement for the l10n-import workflow on firefox-android:
https://github.com/mozilla-mobile/firefox-android/blob/main/.github/workflows/import-l10n.yml
https://github.com/mozilla-l10n/android-l10n/blob/main/.github/scripts/import_strings.py
and also for the l10n sync workflows:
https://github.com/mozilla-mobile/firefox-android/blob/main/.github/workflows/ac-sync-strings.yml
https://github.com/mozilla-mobile/firefox-android/blob/main/.github/workflows/focus-sync-strings.yml
https://github.com/mozilla-mobile/firefox-android/blob/main/.github/workflows/fenix-sync-strings.yml
https://github.com/mozilla-mobile/sync-strings-action/blob/main/src/sync-strings.py

### CODE REVIEW

**Comment by ahal** [comment]
This has substantial conflicts with my treescript refactor just fyi. I'm hoping to get mine landed sometime next week, not sure what the timeline is for this, but you might want to rebase on top of my work sooner rather than later if you think it'll be longer than that.

**Comment by gbrownmozilla** [comment]
A local, manual run with:
```
    "source_repo": "https://hg.mozilla.org/projects/oak",
    "android_l10n_import_info": {
      "from_repo_url": "https://github.com/mozilla-l10n/android-l10n",
      "toml_info": [
        { 
          "toml_path": "mozilla-mobile/fenix/l10n.toml",
          "dest_path": "mobile/android/fenix"
        },
        {
          "toml_path": "mozilla-mobile/focus-android/l10n.toml",
          "dest_path": "mobile/android/focus-android"
        },
        {
          "toml_path": "mozilla-mobile/android-components/l10n.toml",
          "dest_path": "mobile/android/android-components"
        }
      ]
    },
```
generates a sensible looking set of string updates.

I can't think of a straight-forward way to set up a similar test for the merge action, until firefox-android is on both central and beta; re-visit immediately after firefox-android merges on beta?

File: treescript/src/treescript/gecko/__init__.py
Comment (jcristau) [inline treescript/src/treescript/gecko/__init__.py]: Missing android_l10n_merge (and test coverage)?

File: treescript/src/treescript/gecko/android_l10n.py
Comment (jcristau) [inline treescript/src/treescript/gecko/android_l10n.py]: Can we avoid conditionals here?  E.g. pass the `get_android_l10n_*_info` as an argument instead of the `is_import` boolean, and similar below, extract the different paths out of this function.

File: treescript/src/treescript/gecko/android_l10n.py
Comment (jcristau) [inline treescript/src/treescript/gecko/android_l10n.py]: s/merge/sync/ maybe?  That matches the current action's naming and arguably is closer to what is happening.

File: treescript/src/treescript/gecko/mercurial.py
Comment (jcristau) [inline treescript/src/treescript/gecko/mercurial.py:140]: nit: consider pulling this refactor out into a separate commit

File: treescript/src/treescript/gecko/l10n.py
Comment (jcristau) [inline treescript/src/treescript/gecko/l10n.py:188]: nit: consider pulling this refactor out into a separate commit

File: treescript/tests/test_util_treestatus.py
Comment (jcristau) [inline treescript/tests/test_util_treestatus.py]: unused?

**Review by jcristau — CHANGES_REQUESTED** [review]

File: treescript/src/treescript/gecko/android_l10n.py
Comment (gbrownmozilla) [reply treescript/src/treescript/gecko/android_l10n.py]: Good idea. I also considered "uplift" but I think I like "sync" best.

File: treescript/src/treescript/gecko/android_l10n.py
Comment (gbrownmozilla) [reply treescript/src/treescript/gecko/android_l10n.py]: We can but it gets tricky and introduces a lot of parameters. I'll give it a try but I'm also considering separating import and sync entirely (removing android_l10n_action).

File: treescript/src/treescript/gecko/mercurial.py
Comment (gbrownmozilla) [reply treescript/src/treescript/gecko/mercurial.py:140]: Sorry, I should have done that originally. Is it worth doing retroactively?

Comment (jcristau) [reply treescript/src/treescript/gecko/mercurial.py:140]: Happy to leave that up to you.

File: treescript/src/treescript/gecko/android_l10n.py
Comment (jcristau) [inline treescript/src/treescript/gecko/android_l10n.py]: Maybe update the log message to confirm *which* tree is closed.

File: treescript/src/treescript/util/task.py
Comment (jcristau) [inline treescript/src/treescript/util/task.py]: s/merge/sync/

File: treescript/src/treescript/util/task.py
Comment (jcristau) [inline treescript/src/treescript/util/task.py]: s/merge/sync/

File: treescript/src/treescript/util/task.py
Comment (jcristau) [inline treescript/src/treescript/util/task.py]: "tag info" only applies to `get_tag_info` but looks to have been copy/pasted into all `get_*_info` functions.

File: treescript/src/treescript/gecko/android_l10n.py
Comment (jcristau) [inline treescript/src/treescript/gecko/android_l10n.py]: Maybe just `a=l10n` like we're doing for desktop bumps.

File: treescript/tests/test_gecko_android_l10n.py
Comment (jcristau) [inline treescript/tests/test_gecko_android_l10n.py:47]: Should there be a test where get_android_l10n_files_toml does return something?

**Review by jcristau — APPROVED** [review]
Thanks!

File: treescript/src/treescript/gecko/android_l10n.py
Comment (gbrownmozilla) [reply treescript/src/treescript/gecko/android_l10n.py]: Good idea. Thanks.

File: treescript/tests/test_gecko_android_l10n.py
Comment (gbrownmozilla) [reply treescript/tests/test_gecko_android_l10n.py:47]: I've added some additional test cases. Thanks.

**Pull request merged**
